assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/index.js

```diff
@@ -69,6 +139,30 @@ function DashboardAllTrafficWidget() {
 
 	const serviceReportURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getServiceReportURL( reportType, reportArgs ) );
 
+	useEffect( () => {
+		if ( dateRange !== currentRange ) {
+			setFirstLoad( true );
+			setCurrentRange( dateRange );
+		} else if ( pieChartLoaded && totalUsersLoaded && userCountGraphLoaded ) {
+			setFirstLoad( false );
+		}
+	}, [
+		pieChartLoaded,
+		totalUsersLoaded,
+		userCountGraphLoaded,
+		dateRange,
+		currentRange,
+	] );
+
+	if ( pieChartError ) {
+		return <ReportError moduleSlug="analytics" error={ pieChartError } />;
+	}
+
+	const totalUsers = parseInt( pieChartReport?.[ 0 ]?.data?.totals?.[ 0 ]?.values?.[ 0 ] );
+	if ( pieChartLoaded && ( Number.isNaN( totalUsers ) || totalUsers < 1 ) ) {
+		return <ReportZero moduleSlug="analytics" />;
```

Comment:
This should in principle use `props.WidgetReportZero` now instead since #2252 has been merged. However, we'll still need to use this component in the legacy UI where it doesn't receive these props, so it can remain as is for now.

However, let's add a comment:
```
// TODO: Replace with `props.WidgetReportZero` once legacy UI has been removed.
```